extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -191,45 +217,38 @@ void createBeans(FlywayRecorder recorder,
                             hasMigrations, createPossible));
 
             if (DataSourceUtil.isDefault(dataSourceName)) {
-                configurator.addQualifier(Default.class);
+                flywayConfigurator.addQualifier(Default.class);
             } else {
                 String beanName = FLYWAY_BEAN_NAME_PREFIX + dataSourceName;
-                configurator.name(beanName);
-
-                configurator.addQualifier().annotation(DotNames.NAMED).addValue("value", beanName).done();
-                configurator.addQualifier().annotation(FlywayDataSource.class).addValue("value", dataSourceName).done();
+                flywayConfigurator.name(beanName);
+                flywayConfigurator.addQualifier().annotation(DotNames.NAMED).addValue("value", beanName).done();
+                flywayConfigurator.addQualifier().annotation(FlywayDataSource.class).addValue("value", dataSourceName).done();
             }
 
-            syntheticBeanBuildItemBuildProducer.produce(configurator.done());
+            syntheticBeanBuildItemBuildProducer.produce(flywayConfigurator.done());
+
         }
     }
 
     @BuildStep
     @Consume(BeanContainerBuildItem.class)
     @Record(ExecutionTime.RUNTIME_INIT)
-    public ServiceStartBuildItem startActions(FlywayRecorder recorder,
-            FlywayRuntimeConfig config,
+    public void startActions(FlywayRecorder recorder,
+            MigrationStateBuildItem migrationsBuildItem,
+            List<InitTaskCompletedBuildItem> completedInitTasks,
             BuildProducer<JdbcDataSourceSchemaReadyBuildItem> schemaReadyBuildItem,
-            BuildProducer<InitTaskCompletedBuildItem> initializationCompleteBuildItem,
-            MigrationStateBuildItem migrationsBuildItem) {
+            BuildProducer<ServiceStartBuildItem> serviceStart) {
 
-        recorder.doStartActions();
-
-        // once we are done running the migrations, we produce a build item indicating that the
-        // schema is "ready"
-        schemaReadyBuildItem.produce(new JdbcDataSourceSchemaReadyBuildItem(migrationsBuildItem.hasMigrations));
-        initializationCompleteBuildItem.produce(new InitTaskCompletedBuildItem("flyway"));
-        return new ServiceStartBuildItem("flyway");
+        if (completedInitTasks.stream().anyMatch(c -> c.getName().equals(FLYWAY_INIT_TASK))) {
```

Comment:
So, in addition to checking if the job is enabled, we need to check if the tasks have been listed?

Reply:
We need to check that the init thing is completed, before popullating the `ServiceStartBuildItem`

This is what ensures that application will only start after init tasks have been completed.

ApplicationStartBuildItem <- ServiceStart <- InitTaskCompletedBuildItem.

Reply:
So, it's now up to the extensions to track everything. With Kafka, it's annoying as the extension is used in several places that then would need to do that check.